os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -431,6 +435,15 @@ void up_assert(const uint8_t *filename, int lineno)
 
 	board_led_on(LED_ASSERTION);
 
+#ifdef CONFIG_CPP_HAVE_VARARGS
+	va_list valist;
+
+	/* Extract the extra debug arguments if any */
+	va_start(valist, lineno);
+	const char *assertInfo = va_arg(valist, const char *);
```

Comment:
Hi Vidisha, 
There is an issue with this approach. As of now, the `ASSERT` macro is used many places in our code and we pass only a single condition as a parameter to it. So, the variable arguments list in `up_assert()` will be null. In this situation, if we call `va_arg()`, the behaviour is `undefined`.

So, it is better to create a new assert macro to handle the variable arguments.
Please try to change as follows:

Change existing assert macros as below
`#define ASSERT(f)      { if (!(f)) up_assert((const uint8_t *)__FILE__, (int)__LINE__, ""); }`

Create new macro as :
`#define ASSERT_INFO(f, fmt, ...)     { if (!(f)) up_assert((const uint8_t *)__FILE__, (int)__LINE__, fmt,  ##__VA_ARGS__); }`

Change the `up_assert()` as follows:
```
#ifdef CONFIG_CPP_HAVE_VARARGS
void up_assert(const uint8_t *filename, int lineno, char *fmt, ...)
#else
void up_assert(const uint8_t *filename, int lineno, char *fmt)
#endif
{

    lldbg("Assertion failed at file:%s line: %d task: %s\n", filename, lineno, fault_tcb->name);
    if (fmt) {
        va_list valist;
	/* Extract the extra debug arguments if any */
	va_start(valist, fmt);
        lldbg(fmt, valist);
        va_end(valist);
    }
```

Then, from now on, we can use the new `ASSERT_INFO` macro where ever we need it as follows:
`ASSERT_INFO(sem->semcount < SEM_VALUE_MAX, "Sem = 0x%0x count = %d MAX = %d\n", sem, sem->semcount, SEM_VALUE_MAX)`

Reply:
@sunghan-chang Does this approach sound fine ?

Reply:
Hi Kishore,
As discussed over knox, since this approach of using variable arguments has issues with syscall, let's find another suitable approach for this requirement. Till then we have kept the PR as *do not merge*.